fix: constructor LLM should not skip loading other config models

jeffreyscarpenter opened the pull request:
## Description
This PR addresses an issue where the configuration for other models (e.g., content safety, topic control) was not being honored when a main model was provided via the LLMRails (or RunnableRails) constructor. The changes ensure that all model configurations are properly initialized and accessible, even when a main model is injected.

## Related Issue(s)
Fixes #1220 

## Checklist

- [x] I've read the [CONTRIBUTING](https://github.com/NVIDIA/NeMo-Guardrails/blob/develop/CONTRIBUTING.md) guidelines.
- [ ] I've updated the documentation if applicable.
- [ ] I've added tests if applicable.
- [ ] @mentions of the person or team responsible for reviewing proposed changes.

tgasser-nv commented on nemoguardrails/rails/llm/llmrails.py:
```diff
@@ -362,11 +362,14 @@ def _init_llms(self):
+        # treat it as the *main* model, but **still** iterate through the
+        # configuration to load any additional models (e.g. `content_safety`).
+
+        injected_main_llm = False
```
Do we need the `injected_main_llm` boolean, or is this the same as checking `if self.llm`?

tgasser-nv commented on nemoguardrails/rails/llm/llmrails.py:
```diff
@@ -405,12 +408,17 @@ def _init_llms(self):
-                    self.runtime.register_action_param("llm", self.llm)
+                if llm_config.type == "main":
+                    # If a main LLM was already injected, skip creating another
+                    # one. Otherwise, create and register it.
```
Could you log a warning if both the injected and RailsConfig LLM are provided, saying which of the two will be used?

tgasser-nv reviewed (changes requested):
Requested a couple of changes, please also add tests to make sure the priority of the injected and RailsConfig main LLM

jeffreyscarpenter commented:
Thank you @Pouyanpi I have tried to address all the feedback, let me know.

> * the test changes, I noticed you removed a bunch of expected events from test_1 (the "ask if user happy" flow). Is this intentional?

Deleted inadvertently, I have restored them.

> * The `self.main_llm` vs `self.llm` thing, I see you introduced `self.main_llm` but we're still using `self.llm` everywhere else.

Removed all references to `main_llm`

> also it'd be great if you can run pre-commits here and once the pipeline is executed have a look at the coverage report 👍🏻

Ran the pre-commits

Pouyanpi reviewed (approved):
Thank you @jeffreyscarpenter I made some changes, some tests were failing because they never actually tried to initialize their respective model prior to your fix; the early return prevented it, thanks for the fix. We are good to merge 👍🏻
